client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
+	Object.assign( {}, action, { meta: Object.assign( {}, action.meta, analytics.meta ) } );
+
+export const bumpStat = ( group, name ) => ( {
+	type: BUMP_STAT,
+	meta: {
```

Comment:
I overlooked the fact that `redux-analytics` uses `meta: { analytics: { … } }` and have now revised the Pr for that. I'm no longer worried about closing in `meta`, as I think we can probably overwrite the `analytics` property.

Maybe it would still be safer to turn the `analytics` part into an array so that we can continue to compose…

```js
action = compose(
	withAnalytics( bumpStats( 'name', 'value' ) ),
	withAnalytics( recordPageView( 'path', 'title' ) ),
	selectThingAMaBob
);
```

Note: I have auto-curried this function to make these kinds of chains a bit easier